parse/parse.go

```diff
@@ -52,22 +51,46 @@ func GoListAgnostic(stdIn io.Reader) (deps types.ProjectList, err error) {
 	decoder := json.NewDecoder(strings.NewReader(string(johnnyFiveNeedInput)))
 
 	for {
-		var mod types.GoListModule
+		var mod map[string]interface{}
 
-		err = decoder.Decode(&mod)
-		if err == io.EOF {
-			err = nil
+		decodeErr := decoder.Decode(&mod)
+
+		if decodeErr == io.EOF {
 			break
 		}
-		if err != nil {
+		if decodeErr != nil {
+			err = decodeErr
 			break
 		}
 
-		project, err := modToProjectList(mod)
-		if _, ok := err.(*NoVersionError); ok {
+		if module, ok := mod["Module"].(map[string]interface{}); ok {
+			if version, ok := module["Version"].(string); ok {
+				deps.Projects = append(deps.Projects, types.Projects{Version: version, Name: module["Path"].(string)})
```

Comment:
I was iffy on doing another type assertion check on `module["Path"]` because I know if `Version` is there then `Path` is but if anyone wants me to go through the extra hoop let me know!

Reply:
I'm less interested in the extra hoop, but some code comments about the "why" might illuminate.